GitHubBranches.ps1

```diff
@@ -104,6 +135,48 @@ function Get-GitHubRepositoryBranch
         'NoStatus' = (Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus -ConfigValueName DefaultNoStatus)
     }
 
-    return Invoke-GHRestMethodMultipleResult @params
+    return (Invoke-GHRestMethodMultipleResult @params | Add-GitHubBranchAdditionalProperties)
 }
 
+filter Add-GitHubBranchAdditionalProperties
+{
+<#
+    .SYNOPSIS
+        Adds type name and additional properties to ease pipelining to GitHub Branch objects.
+
+    .PARAMETER InputObject
+        The GitHub object to add additional properties to.
+
+    .PARAMETER TypeName
+        The type that should be assigned to the object.
+#>
+    [CmdletBinding()]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseSingularNouns", "", Justification="Internal helper that is definitely adding more than one property.")]
```

Comment:
Yeah...I do keep a ruler/guideline in my editor at the 100 character mark and try to keep everything under that (manually wrapped).  Those rule suppression lines tended to be my main exception to that just because it's not an integral part of the code, and I thought that breaking that out across multiple lines-per-rule ended up reducing the overall glanceability/readability of the code overall.  Will consider it, but that's an exception I'm likely to keep.